crates/redis_interface/src/commands.rs

```diff
@@ -852,12 +852,28 @@ impl super::RedisConnectionPool {
             .await
             .change_context(errors::RedisError::ConsumerGroupClaimFailed)
     }
+
+    #[instrument(level = "DEBUG", skip(self))]
+    pub async fn set_keys_using_script(
+        &self,
+        keys: Vec<String>,
+        values: Vec<String>,
```

Comment:
Have we evaluated the case if `keys` and `values` have different length?
in that case, better write this as `key_values: Vec<(String, usize)>`